File: tests/adapters/go/goldens/complex/max_optimization.go
// Comprehensive Go sample for Budget System tests.
package main

// … comment omitted
import (
	// … 3 imports omitted
)

// … comment omitted
import (
	// … import omitted
)

// … comment omitted
import (
	// … 2 imports omitted
)

// ModuleTitle is a module level long documentation that might be truncated under tight budgets.
const ModuleTitle = "Budget System Complex Sample"

const LongText = `This is an extremely long text that is designed to be trimmed
by the literal optimizer when budgets are small. It repeats a message to
ensu…` // literal string (−15 tokens)

var BigObject = map[string]interface{}{
	"users": func() []map[string]interface{} {
		result := make([]map[string]interface{}, 50)
		for i := 0; i < 50; i++ {
			result[i] = map[string]interface{}{
				"id":     i + 1,
				// … (2 more, −23 tokens)
			}
		}
		return result
	}(),
	// … (1 more, −128 tokens)
}

// PublicService provides public API operations.
type PublicService struct {
	// … field omitted
}

// NewPublicService creates a new service instance.
func NewPublicService() *PublicService // … function body omitted (5 lines)

// … comment omitted
func (s *PublicService) GetUser(id int) *models.User // … method body omitted (7 lines)

// … comment omitted
// … method omitted

// … comment omitted
func (s *PublicService) Process(list []*models.User) ([]*models.User, error) // … method body omitted (15 lines)

// … comment omitted
// … struct omitted (3 lines)

// … method omitted

// PublicFunction is an exported function.
func PublicFunction(name string) string // … function body omitted (7 lines)

// … 2 functions omitted

File: tests/adapters/java/goldens/complex/max_optimization.java
/**
 * Comprehensive Java sample for Budget System tests.
 */

package com.example.budget;

// … comment omitted
// … 4 imports omitted

// … comment omitted
// … 3 imports omitted

/**
 * Module level long documentation that might be truncated under tight budgets.
 */
public class BudgetSystemSample {
    public static final String MODULE_TITLE = "Budget System Complex Sample";

    public static final String LONG_TEXT = """
        This is an extremely long text block that is designed to be trimmed
        by the literal optimizer when budgets are small. It repeats a m…"""; // literal string (−8 tokens)

    public static final Map<String, Object> BIG_OBJECT = Map.ofEntries(
        Map.entry("users", IntStream.range(1, 51)
            .mapToObj(i -> Map.of(
                "id", i,
                "name", "User " + i,
                "active", i % 2 == 0
            ))
            .collect(Collectors.toList()))
        // … (1 more, −86 tokens)
    );
}

public class PublicService {
    // … field omitted

    /**
     * Public API: gets a user by ID.
     */
    public User getUser(long id) // … method body omitted (3 lines)

    // … comment omitted
    private User normalize(Map<String, Object> u) // … method body omitted (7 lines)

    /** Long method body to allow function body stripping. */
    public ApiResponse<List<User>> process(List<User> list) // … method body omitted (12 lines)
}

// … class omitted

public class Functions {
    public static String publicFunction(String name) // … method body omitted (4 lines)

    private static List<String> privateFunction(List<String> data) // … method body omitted (6 lines)

    public static void main(String[] args) // … method body omitted (4 lines)
}

// … comment omitted
// … 3 classes omitted
